src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -2555,9 +2595,16 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 
 		ProblemMatcherRegistry.onReady().then(() => {
 			return this.editorService.saveAll({ reason: SaveReason.AUTO }).then(() => { // make sure all dirty editors are saved
+				// {{SQL CARBON EDIT}}
+				let lastRunTasksViewTask = this.lastRunTasksViewTask;
+				lastRunTasksViewTask.taskId = UUID.generateUuid();
+				this.sqlTaskService.createNewTask(lastRunTasksViewTask);
+				this.lastRunTasksViewTask = lastRunTasksViewTask;
```

Comment:
what does this.sqlTaskService.createNewTask do? according to its name it should return an object, right?
also, you assigned this.lastRunTasksViewTask to a local variable and then assign it back to this.lastRunTasksViewTask, is it really necessary? is createNewTask actually updating the lastRunTasksViewTask variable?

Reply:
Replied in other comment